src/components/Header/index.tsx

```diff
@@ -38,93 +33,70 @@ const Header = ({
   currentTab,
   onChangeTab,
   onToggleMenu,
-  tabs
+  tabs,
+  captureRef
 }: HeaderProps) => {
-  const [{ term, oscar, pinnedCrns }, { setTerm }] = useContext(TermContext);
+  const [
+    { term, versionName, oscar, pinnedCrns },
+    { setTerm, setVersionName }
+  ] = useContext(ScheduleContext);
   const [terms] = useContext(TermsContext);
-  const [theme, setTheme] = useContext(ThemeContext);
-  const captureRef = useRef<HTMLDivElement>(null);
+  const [versionLists, { patchVersionsData }] = useContext(VersionsContext);
+  const versionList = versionLists[term];
+  const [theme] = useContext(ThemeContext);
+  const [versionIndex, setVersionIndex] = useState(
```

Comment:
What is the difference between the versionIndex and position of versionName in the versionLists[term] list?

Reply:
There is no difference. "versionIndex" is  the index of "versionName" in its corresponding version list. In header, "versionIndex" is used to perform various operations to version lists, "versionName" is never used directly except getting the right "versionIndex" at the beginning.

Reply:
When I first thought it, I was worried that it might be a bit of a code smell that the state has been duplicated, but since you say that the versionName is only used initially I think it's fine.